test/l1-resources/api-key/index.test.ts

```diff
@@ -0,0 +1,70 @@
+import { App, Stack } from "aws-cdk-lib";
+import { Template } from "aws-cdk-lib/assertions";
+
+import {
+  CfnApiKey,
+  CfnApiKeyProps,
+  ProjectAssignment,
+  ListOptions,
+} from "../../../src/l1-resources/api-key";
+
+const RESOURCE_NAME = "MongoDB::Atlas::APIKey";
+const ORG_ID = "63350255419cf25e3d511c95";
+const PROFILE = "default";
+const Roles = ["ORG_MEMBER", "ORG_GROUP_CREATOR"];
+const proAssignments: ProjectAssignment[] = [
+  {
+    projectId: "64ee3fcfaa95ad6b7e358d56",
+    roles: ["GROUP_READ_ONLY"],
+  },
+  {
+    projectId: "64ee3fd58bb58153335e8469",
+    roles: ["GROUP_SEARCH_INDEX_EDITOR", "GROUP_OWNER", "GROUP_READ_ONLY"],
+  },
+];
+
+const listOptions: ListOptions = {
+  pageNum: 1,
+  itemsPerPage: 300,
+};
+
+const propsQueryLimitProps: CfnApiKeyProps = {
+  orgId: ORG_ID,
+  description: "for cfn test api-key resource testing.",
+  profile: PROFILE,
+  roles: Roles,
+  awsSecretName: "mongodb/atlas/apikey/cfn-bot-apikey-test",
+  projectAssignments: proAssignments,
+  listOptions: listOptions,
+};
+
+test("CfnApiKey construct should contain default properties", () => {
```

Comment:
is it possible to generally add more tests? would be nice to also cover paths like missing fields / wrong field's format / etc.

Reply:
i dont think is a good idea to go further on tests, the CFN resource is already deeply tested, and having more tests in both cfn and cdk would make a maintenance nightmare

Reply:
> would make a maintenance nightmare

can you clarify where the nightmare comes from? I am genuinely curious because I am new to the code base.

Also when you say:

> the CFN resource is already deeply tested

where is this being deeply tested? Are you referring to tests within CFN itself?

Reply:
yes! the cfn test, within the cloud formation resource, and when you mention formatting, we defined some formatting with regex on the cloud formation resource, but i ended removing the regex con the CLOUD FORMATION resource, becuase everytime something changed on the client, we had to change the cloud formation resource and redeploy it, so no we are letting the atlas client validate the format on the fields

Reply:
Thank you for the explanation, even though I am not sure it relates to the original point of unit test coverage. 

Let's set-up some time for a call to go through the code base